libflux/src/flux/semantic/flatbuffers/mod.rs

```diff
@@ -1,3 +1,1150 @@
 #![allow(clippy::all)]
 pub mod semantic_generated;
 pub mod types;
+
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use crate::ast;
+
+use crate::semantic;
+use crate::semantic::walk;
+use flatbuffers::{UnionWIPOffset, WIPOffset};
+use semantic_generated::fbsemantic;
+
+extern crate chrono;
+use chrono::Offset;
+
+pub fn serialize(semantic_pkg: &mut semantic::nodes::Package) -> Result<(Vec<u8>, usize), String> {
+    let mut v = new_serializing_visitor_with_capacity(1024);
+    walk::walk(&mut v, Rc::new(walk::Node::Package(semantic_pkg)));
+    v.finish()
+}
+
+fn new_serializing_visitor_with_capacity<'a>(_capacity: usize) -> SerializingVisitor<'a> {
+    SerializingVisitor {
+        inner: Rc::new(RefCell::new(SerializingVisitorState::new_with_capacity(
+            1024,
+        ))),
+    }
+}
+
+struct SerializingVisitor<'a> {
+    inner: Rc<RefCell<SerializingVisitorState<'a>>>,
+}
+
+impl<'a> semantic::walk::Visitor<'_> for SerializingVisitor<'a> {
+    fn visit(&mut self, _node: Rc<walk::Node<'_>>) -> bool {
+        let v = self.inner.borrow();
+        if let Some(_) = &v.err {
+            return false;
+        }
+        Some(SerializingVisitor {
+            inner: Rc::clone(&self.inner),
+        });
+        true
+    }
+
+    fn done(&mut self, node: Rc<walk::Node<'_>>) {
+        let mut v = &mut *self.inner.borrow_mut();
+        if let Some(_) = &v.err {
+            return;
+        }
+        let node = &*node;
+        let loc = v.create_loc(node.loc());
+        match node {
+            walk::Node::IntegerLit(int) => {
+                let int_typ = int.typ.clone();
```

Comment:
I don't think we should need to always clone the type to serialize it.  I think `build_type` should be made to accept a reference.  If this isn't an easy fix, can you file a bug?

Reply:
Yeah I'll file a bug for this. I played around with fixing it. Doesn't seem like a ton of work, but it does cascade into a few other bugs a bit.

Reply:
https://github.com/influxdata/flux/issues/2292